### DISCUSSION TITLE
Why does Term::map_from_pairs return a NifResult?

### DISCUSSION THREAD

**Discussion by radrow** [rusterlium/rustler#554] [Q&A] [answered]
I don't understand the circumstances in which it fails. Creating a new map is safe, so should be inserting to something we know is a map. Am I missing something?

**Comment by filmor** [comment]
It fails on duplicate keys.

**Reply by radrow** [reply]
Why though? It's the first time I see this considered to be a reason to fail to be honest. Also, Erlang does not crash if you do it.

**Reply by filmor** [reply, marked as answer]
The function just fails, nothing crashes. See https://www.erlang.org/doc/man/erl_nif.html#enif_make_map_from_arrays, which is the underlying function that is exposed here.
